### PR TITLE
Added update/delete camera share request endpoint

### PR DESCRIPTION
**Pull request by azharmalik3**
@mosic

### CODE REVIEW

**Comment by mosic** [comment]
@azharmalik3 Nice work. The PR is almost ready to merge, it would just be better to use Ecto Changeset for parameter validation. Consult with @ijunaidfarooq and let me know if you have questions about this.

**Comment by azharmalik3** [comment]
@mosic added all validation in changeset.

File: web/models/camera_share_request.ex [hunk 1]
```diff
@@ -33,11 +38,46 @@ defmodule CameraShareRequest do
...
     case status do
       "used" -> @status.used
       "cancelled" -> @status.cancelled
       "pending" -> @status.pending
     end
   end
+
+  defp validate_rights(changeset, field) do
+    rights = get_field(changeset, field)
+    if rights do
```
Comment (mosic) [inline web/models/camera_share_request.ex]: Simplify this using `with`.

File: web/controllers/camera_share_request_controller.ex [hunk 2]
```diff
@@ -4,25 +4,70 @@ defmodule EvercamMedia.CameraShareRequestController do
...
+    end
+  end
+
+  defp caller_has_permission(conn, user, camera) do
+    if Permission.Camera.can_edit?(user, camera) do
+      :ok
+    else
+      conn
+      |> put_status(401)
+      |> render(ErrorView, "error.json", %{message: "Unauthorized."})
```
Comment (mosic) [inline web/controllers/camera_share_request_controller.ex]: Use `render_error` instead.

File: web/models/camera_share_request.ex [hunk 3]
```diff
@@ -33,11 +38,46 @@ defmodule CameraShareRequest do
...
+    else
+      add_error(changeset, :rights, "Invalid rights specified in request.")
+    end
+  end
+
+  def changeset(model, params \\ :invalid) do
+    model
+    |> cast(params, @required_fields, @optional_fields)
+    |> validate_required(:email)
+    |> validate_format(:email, ~r/\A.+\@.+\..+\z/, [message: "You've entered an invalid email address."])
```
Comment (mosic) [inline web/models/camera_share_request.ex]: Use the email regex from `User` module.

**Comment by azharmalik3** [comment]
@mosic It's ready to merge.

**Comment by mosic** [comment]
@azharmalik3 This PR is not compatible with https://github.com/evercam/evercam-media/pull/255, which I just merged. Update it to work with master and I'll merge it.

**Comment by azharmalik3** [comment]
@mosic I have rebased branched on top of master and also made some changes in `share_request` model for insert and update.

**Comment by mosic** [comment]
@azharmalik3 Good work, rebased to master. I refactored some of the code to reduce duplication in https://github.com/evercam/evercam-media/commit/25d64dd1bdb262024b3848e560e587ea5ff9ae7f.

I also renamed the controller action to `cancel` (from `delete`), as the old name was misleading - it cancels the share request, it doesn't delete it.

**Pull request closed without merging**